src/component/common/ConstraintAccordion/ConstraintAccordionEdit/ConstraintAccordionEditBody/useConstraintInput/useConstraintInput.tsx

```diff
@@ -46,202 +78,225 @@ export const useConstraintInput = ({
     setCaseInsensitive,
     removeValue,
 }: IUseConstraintInputProps): IUseConstraintOutput => {
-    const [input, setInput] = useState<JSX.Element>(<></>);
-    const [validator, setValidator] = useState(() => {
-        return () => false;
-    });
+    const [input, setInput] = useState<Input>(IN_OPERATORS_LEGAL_VALUES);
+    const [validator, setValidator] = useState<Validator>(
+        STRING_ARRAY_VALIDATOR
+    );
     const [error, setError] = useState('');
 
-    const resolveInputType = useCallback(() => {
```

Comment:
Could this be a regular component outside the hook (that takes an `Input` prop)?